go/private/rules/transition.bzl

```diff
@@ -250,15 +255,44 @@ def _go_reset_transition_impl(settings, attr):
         settings[filter_transition_label(label)] = value
     return settings
 
-go_reset_transition = transition(
-    implementation = _go_reset_transition_impl,
+go_exec_reset_transition = transition(
+    implementation = _go_exec_reset_transition_impl,
     inputs = _reset_transition_keys,
     outputs = _reset_transition_keys,
 )
 
-def _go_reset_target_impl(ctx):
+_non_go_exec_reset_transition_dict = dict(_common_reset_transition_dict, **{
+    "@io_bazel_rules_go//go/private:bootstrap_nogo": False,
+})
+
+_non_go_exec_reset_transition_keys = sorted([filter_transition_label(label) for label in _non_go_exec_reset_transition_dict.keys()])
```

Comment:
`_non_go_exec_reset_transition_keys` can be constructed from `_reset_transition_dict` too, because it as the same set of keys as `_non_go_exec_reset_transition_dict`. If we do that, then we can don't need `_non_go_exec_reset_transition_dict`, and instead setting `@io_bazel_rules_go//go/private:bootstrap_nogo` in   `_go_non_go_exec_reset_tool_transition_impl`

Reply:
I ended up removing `_non_go_exec_reset_transition_keys` as well in favor of `_reset_transition_keys`.